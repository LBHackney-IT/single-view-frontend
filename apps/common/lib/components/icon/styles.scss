.mtfh-icon {
  display: inline-block;
  color: currentColor;
  line-height: 1em;
  flex-shrink: 0;
  vertical-align: middle;
}
