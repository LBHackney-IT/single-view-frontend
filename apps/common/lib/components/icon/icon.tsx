import React, { ComponentPropsWithoutRef, forwardRef } from "react";

import cn from "classnames";

import "./styles.scss";

export interface IconProps extends ComponentPropsWithoutRef<"svg"> {
  viewBox: string;
  size?: string;
  color?: string;
}

export const Icon = forwardRef<SVGSVGElement, IconProps>(function Icon(
  { width, height, color, className, size = "1em", focusable = false, ...props },
  ref,
) {
  const iconClasses = cn("mtfh-icon", className);

  const style: Record<string, string | number> = {};

  if (color && color !== "currentColor") {
    style.color = color;
  }

  return (
    <svg
      ref={ref}
      className={iconClasses}
      width={width || size}
      height={height || size}
      focusable={focusable}
      style={style}
      {...props}
    />
  );
});
