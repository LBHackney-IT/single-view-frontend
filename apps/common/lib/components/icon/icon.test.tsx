import React, { useEffect } from "react";

import { render, testA11y } from "@hackney/mtfh-test-utils";

import { Icon } from "./icon";

test("it renders a icon with defaults", async () => {
  const { container } = render(
    <Icon viewBox="0 0 200 200">
      <path
        fill="currentColor"
        d="M 100, 100 m -75, 0 a 75,75 0 1,0 150,0 a 75,75 0 1,0 -150,0"
      />
    </Icon>,
  );
  expect(container).toMatchSnapshot();
  await testA11y(container);
});

test("it renders a icon with inline color", () => {
  const { container } = render(
    <Icon viewBox="0 0 200 200" color="red">
      <path
        fill="currentColor"
        d="M 100, 100 m -75, 0 a 75,75 0 1,0 150,0 a 75,75 0 1,0 -150,0"
      />
    </Icon>,
  );
  expect(container).toMatchSnapshot();
});

test("it accepts a ref", () => {
  const callback = jest.fn();
  const Comp = () => {
    const ref = React.createRef<SVGSVGElement>();

    useEffect(() => {
      if (ref.current) {
        callback(ref.current);
      }
    }, [ref]);
    return (
      <Icon ref={ref} viewBox="0 0 200 200">
        <path
          fill="currentColor"
          d="M 100, 100 m -75, 0 a 75,75 0 1,0 150,0 a 75,75 0 1,0 -150,0"
        />
      </Icon>
    );
  };
  render(<Comp />);
  expect(callback).toBeCalledTimes(1);
});
