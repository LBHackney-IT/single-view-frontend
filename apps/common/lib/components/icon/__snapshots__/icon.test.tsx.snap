// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders a icon with defaults 1`] = `
<div>
  <svg
    class="mtfh-icon"
    focusable="false"
    height="1em"
    viewBox="0 0 200 200"
    width="1em"
  >
    <path
      d="M 100, 100 m -75, 0 a 75,75 0 1,0 150,0 a 75,75 0 1,0 -150,0"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`it renders a icon with inline color 1`] = `
<div>
  <svg
    class="mtfh-icon"
    focusable="false"
    height="1em"
    style="color: red;"
    viewBox="0 0 200 200"
    width="1em"
  >
    <path
      d="M 100, 100 m -75, 0 a 75,75 0 1,0 150,0 a 75,75 0 1,0 -150,0"
      fill="currentColor"
    />
  </svg>
</div>
`;
